test/unit/collection-view/collection-view-children.spec.js

```diff
@@ -204,6 +204,93 @@ describe('CollectionView Children', function() {
       this.sinon.spy(myCollectionView, 'sort');
     });
 
+    describe('when called with preventRender option', function() {
+
+      beforeEach(function() {
+        this.sinon.spy(myCollectionView, '_renderChildren');
```

Comment:
Mn does not test private functions.  So we need to test public API / hooks _or_ in some cases public to the module, but not necessarily exposed.  ie: `children._add` is considered public to the module, but not exposed for end users.  `_renderChildren` and flags like `_addingMultipleIndexedViews` should not be tested.  We should be free to modify implementation details without breaking tests. Tests should only break if a major version of Marionette is required.

Reply:
yes, i was indoubt about this.
but can not realize yet should i just delete this two test or replace it with some else
for renderChildren maybe its possibe just to check onRenderChildren
but what about _addingMultipleIndexedViews 
does this mean that implementation is not good enough?

Reply:
correct I think testing for render events is fine.  Regarding the flag, it's important we test what the flag is doing and not how the flag is implemented.  Whether the flag is set or deleted is unimportant.  We need to test the scenarios when using `addChildView` that setting or deleting the flag cover.  However as mentioned elsewhere I think we should remove the `_addedViews` perf when using `preventRender` as if a render is not immediately called, subsequent other renders may break.  We _can_ however after this is implemented probably account for the perf bonus if we end up supporting an array of views.

Reply:
i think that this can solve this issue
```
    // Only cache views if added to the end
    if (!hasIndex || index >= this._children.length) {
      this._addedViews || (this._addedViews = []);
      this._addedViews.push(view);
    }
```
last time you tell me that _addedViews not intended for continuos operations
but maybe in this scenario its possible to do it like this?

Reply:
I think attempting to maintain `_addedViews` is a very very bad idea with many caveats.
```js
myCollectionView.addChildView(new View(), { preventRender: true }); // this._addedViews exists with new view
myCollectionView.collection.add(fooModel); // this._addedViews exists so new view gets attached, fooModel's view does not
myCollectionView.addChildView(new View()); // new view gets added immediately.. still no fooModel view..
```